In [ ]:
import pandas as pd
import numpy as np

df = pd.read_csv('/content/enem_simulado_brasil.csv')
df.head()


,id_estudante,estado,cidade,tipo_escola,renda_familiar,nota_matematica,nota_linguagens,nota_ciencias,nota_humanas,nota_redacao,media_final
0,1,RJ,Rio de Janeiro,Privada,4277.46,595.34,648.40,699.21,675.79,682.02,660.15
1,2,AM,Manaus,Privada,10758.81,788.20,797.79,648.60,759.84,795.12,757.91
2,3,BA,Feira de Santana,Pública,11641.62,730.29,636.45,704.96,721.38,685.06,695.63
3,4,RJ,Duque de Caxias,Pública,4963.13,391.00,493.05,508.04,434.84,375.36,440.46
4,5,RJ,Niterói,Privada,6197.79,452.66,576.27,476.36,426.11,601.69,506.62


In [ ]:
df.shape
df.columns
df.info()
df.describe()
df.isnull().sum()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   id_estudante     1000 non-null   int64  
 1   estado           1000 non-null   object 
 2   cidade           1000 non-null   object 
 3   tipo_escola      1000 non-null   object 
 4   renda_familiar   1000 non-null   float64
 5   nota_matematica  1000 non-null   float64
 6   nota_linguagens  1000 non-null   float64
 7   nota_ciencias    1000 non-null   float64
 8   nota_humanas     1000 non-null   float64
 9   nota_redacao     1000 non-null   float64
 10  media_final      1000 non-null   float64
dtypes: float64(7), int64(1), object(3)
memory usage: 86.1+ KB


,0
id_estudante,0
estado,0
cidade,0
tipo_escola,0
renda_familiar,0
nota_matematica,0
nota_linguagens,0
nota_ciencias,0
nota_humanas,0
nota_redacao,0


In [ ]:
analise_escola = df.groupby('tipo_escola')[['nota_matematica','nota_linguagens','nota_ciencias','nota_humanas','nota_redacao','media_final']].mean()
display(analise_escola)

,nota_matematica,nota_linguagens,nota_ciencias,nota_humanas,nota_redacao,media_final
tipo_escola,,,,,,
Privada,659.234422,641.337381,639.861190,641.487449,639.355782,644.255170
Pública,613.824207,588.204490,596.248499,593.907606,595.636771,597.564263


In [ ]:
media_estado = df.groupby('estado')['media_final'].mean().sort_values(ascending=False)
display(media_estado.head())

top10 = df.sort_values('media_final', ascending=False).head(10)
display(top10)

,media_final
estado,
SP,629.610465
PR,619.856111
RJ,618.105179
DF,612.376000
CE,612.243077


,id_estudante,estado,cidade,tipo_escola,renda_familiar,nota_matematica,nota_linguagens,nota_ciencias,nota_humanas,nota_redacao,media_final
186,187,SP,Campinas,Privada,14461.50,968.43,1000.00,966.94,1000.00,803.48,947.77
441,442,SP,São Paulo,Pública,12543.54,950.64,778.02,920.15,793.57,823.48,853.17
790,791,AM,Itacoatiara,Privada,11850.24,839.86,768.65,897.04,886.51,755.56,829.52
998,999,BA,Vitória da Conquista,Pública,12462.11,865.98,843.98,881.28,828.07,724.61,828.79
746,747,RJ,Nova Iguaçu,Privada,3350.79,726.35,782.37,824.06,830.19,924.65,817.53
677,678,MG,Contagem,Privada,1728.36,818.08,842.27,725.77,840.52,843.25,813.98
267,268,CE,Juazeiro do Norte,Pública,14556.15,920.30,717.34,777.47,835.27,791.99,808.47
276,277,RJ,Duque de Caxias,Privada,5178.62,796.15,794.60,775.54,776.95,874.65,803.58
921,922,CE,Caucaia,Pública,9029.90,846.58,811.61,729.46,744.86,881.40,802.78
132,133,SP,Santos,Privada,8118.45,810.38,800.86,757.73,740.38,902.31,802.33


In [ ]:
display(df[['nota_matematica','nota_linguagens','nota_ciencias','nota_humanas','nota_redacao']].mean())
display(df[['renda_familiar','media_final']].corr(numeric_only=True))

,0
nota_matematica,627.17481
nota_linguagens,603.82556
nota_ciencias,609.07063
nota_humanas,607.89608
nota_redacao,608.49016


,renda_familiar,media_final
renda_familiar,1.000000,0.026982
media_final,0.026982,1.000000


Sim, existe uma diferença. As escolas Privadas apresentaram médias superiores: 644.25, enquanto as escolas Públicas tem: 597.56

Os 5 estados com maior média:
SP (629.61)
PR (619.85)
RJ (618.10)
DF (612.37)
CE (612.24)

Os dados indicam que estudantes de famílias com maior poder aquisitivo tendem a obter melhores resultados

Reforçar escolas públicas, sobretudo em exatas, para reduzir o gap com privadas.
Apoio financeiro para alunos de baixa renda com alto potencial.
Replicar práticas de estados com melhor desempenho (ex.: SP e PR).